commit b56381ebfe0703a082f6218a4316b7fad884f4bc (refs/changes/03/2503/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-09-06T14:15:11-07:00 (2 years, 1 month ago)
    
    DOCKER-1097 docker pull fails for registry server that has no auth setup

diff --git a/CHANGES.md b/CHANGES.md
index f936c5d..7757280 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,10 @@
 
 (nothing yet)
 
+## 3.2.9
+
+- DOCKER-1097 docker pull fails for registry server that has no auth setup
+
 ## 3.2.8
 
 - DOCKER-1091 Cannot login into Artifactory private docker registry. This change
diff --git a/lib/registry-client-v2.js b/lib/registry-client-v2.js
index bdb991f..d4177ec 100644
--- a/lib/registry-client-v2.js
+++ b/lib/registry-client-v2.js
@@ -710,6 +710,8 @@ function ping(opts, cb) {
  *          authInfo    an object with authentication info, examples:
  *                          {type: 'basic', username: '...', password: '...'}
  *                          {type: 'bearer', token: '...'}
+ *                      which can be the empty object when no auth is needed:
+ *                          {}
  */
 function login(opts, cb) {
     assert.object(opts, 'opts');
@@ -773,6 +775,7 @@ function login(opts, cb) {
                     assert.equal(res.statusCode, 200,
                         'ping success without 200');
                     // No authorization is necessary.
+                    authInfo = {};
                     next(true);  // early pipeline abort
                 } else if (res && res.statusCode === 401) {
                     var chalHeader = res.headers['www-authenticate'];
@@ -869,6 +872,7 @@ function login(opts, cb) {
         if (err) {
             cb(err);
         } else {
+            assert.object(authInfo, 'authInfo');
             cb(null, {
                 status: 'Login Succeeded',
                 authInfo: authInfo
diff --git a/package.json b/package.json
index c3dd33d..bfad19f 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "docker-registry-client",
-    "version": "3.2.8",
+    "version": "3.2.9",
     "description": "node.js client for the Docker Registry API",
     "author": "Joyent (joyent.com)",
     "main": "./lib/index.js",
